packages/hyperion-autologging/src/ALInteractableDOMElement.ts

```diff
@@ -182,113 +174,127 @@ const extractInnerText = (element: HTMLElement | null): string | null => {
+function getTextFromElementsByIds(domSource: ALDOMTextSource, source: ALElementText['source']): ALElementText | null {
+  const indirectSources = domSource.directSource
+    .getAttribute(source)
+    ?.split(' ')
+    .map(id => document.getElementById(id))
+    .filter(function (element): element is HTMLElement { return element instanceof HTMLElement });
+  const fullText = indirectSources?.map(extractInnerText).filter(text => text != null).join(' ');
+  if (fullText != null && fullText !== '') {
+    domSource.indirectSources = indirectSources
+    return callExternalTextProcessor(
+      {
+        text: fullText,
+        source,
+      },
+      domSource
+    );
+  }
+  return null;
+}
 
-    const describedBy = nextElement.getAttribute('aria-describedby');
-    if (describedBy != null) {
-      const descText = getTextFromElementsByIds(describedBy);
-      if(descText!= null && descText!== '') {
-        return {
-          text: descText,
-          source: 'aria-describedby',
-        }
-      }
-      // const fullDesc = describedBy.split(' ').map(describedBy => {
-      //   const descElement = document.getElementById(describedBy);
-      //   if (descElement != null) {
-      //     const descText = extractInnerText(descElement);
-      //     if (descText != null) {
-      //       return descText;
+function getTextFromElementAttribute(domSource: ALDOMTextSource, source: ALElementText['source']): ALElementText | null {
+  const label = domSource.directSource.getAttribute(source);
+  if (label != null && label !== '') {
+    return callExternalTextProcessor(
+      {
+        text: label,
+        source
+      },
+      domSource
+    );
+  }
+  return null;
+}
 
-      //     }
-      //   }
-      //   return null;
-      // });
-      // const fullDescText = fullDesc.join(' ');
+function getTextFromInnerText(domSource: ALDOMTextSource, source: ALElementText['source']): ALElementText | null {
+  const text = extractInnerText(domSource.directSource);
+  if (text != null && text !== '') {
+    return callExternalTextProcessor(
+      {
+        text,
+        source
+      },
+      domSource
+    );
+  }
+  return null;
+}
 
-      // if (fullDescText != null && fullDescText !== '') {
-      //   return {
-      //     text: fullDescText,
-      //     source: 'aria-describedby',
-      //   };
-      // }
-    }
+export function getElementName(element: HTMLElement, surface: string | null): ALElementText | null {
+  const domSource: ALDOMTextSource = {
+    directSource: element,
+    surface
+  };
+  for (
+    let nextElement: HTMLElement | null = element;
+    nextElement && nextElement.nodeType === Node.ELEMENT_NODE;
+    nextElement = nextElement.parentElement
+  ) {
 
-    const name = extractInnerText(nextElement);
-    if (name != null && name !== '') {
-      return {text: name, source: 'innerText'};
+    const text = getTextFromElementsByIds(domSource, 'aria-labelledby')
```

Comment:
`nextElement` is not passed to these extraction methods,  looks like we just use the same `directSource: element` each time.

Not sure if we want to add each iteration's (after the first) element as `indirectSources` or not.

Reply:
It might actually be better to not loop, but open to thoughts.   In some cases we may walk way up the dom and get some text that's totally unrelated

Reply:
Oh shoot! This was missing! I should update the `domSource.directSource = nextElement` in each iteration. 

the looping here is necessary because of the priority of different attributes.